vm/env.go

```diff
@@ -422,3 +422,32 @@ func (e *Env) Execute(src string) (interface{}, error) {
 func (e *Env) Run(stmts []ast.Stmt) (interface{}, error) {
 	return Run(stmts, e)
 }
+
+// Copy the state of the virtual machine environment
+func (e *Env) Copy() *Env {
+	b := false
+	copy := Env {
```

Comment:
It doesn't really matter, but would be kind of nice if it was in the same order as the struct declare.
type Env struct {
	name      string
	env       map[string]reflect.Value
	typ       map[string]reflect.Type
	parent    *Env
	interrupt *bool
	external  EnvResolver

Reply:
Reordered.